packages/ui-test-queries/src/utils/bindElementToUtilities.ts

```diff
@@ -25,32 +25,45 @@ import { fireEvent } from './events'
 
 import * as helpers from './helpers'
 import * as queries from './queries'
-import { bindElementToMethods } from './bindElementToMethods'
+import { bindElementToMethods, GenericFunction } from './bindElementToMethods'
 import { bindElementToEvents } from './bindElementToEvents'
 import { isElement } from './isElement'
 
-function bindElementToUtilities(element, customMethods = {}) {
+type QueryTypes = {
+  [K in Extract<keyof typeof queries, string>]: typeof queries[K]
+}
+type HelperTypes = {
+  [K in Extract<keyof typeof helpers, string>]: typeof helpers[K]
+}
+
+export type BoundType = QueryTypes & HelperTypes
+
+function bindElementToUtilities<T>(
+  element: T | T[],
+  customMethods: Record<string, GenericFunction> = {}
+): BoundType[] {
   if (!element) {
-    return element
+    return null as any // just seems to happen deep in the recursion
   } else if (Array.isArray(element)) {
-    return element.map((el) => bindElementToUtilities(el, customMethods))
-  } else if (typeof element.getDOMNode === 'function') {
+    const boundElements = element.map((el) =>
+      bindElementToUtilities(el, customMethods)
+    )
+    return boundElements as any // just seems to happen deep in the recursion
+  } else if (typeof (element as any).getDOMNode === 'function') {
     // eslint-disable-next-line no-param-reassign
-    element = element.getDOMNode()
+    element = (element as any).getDOMNode()
   }
-
   if (!isElement(element)) {
     throw new Error(
       '[ui-test-queries] could not bind utilities to invalid DOM Element!'
     )
   }
-
   return {
     ...bindElementToMethods(element, queries),
-    ...bindElementToEvents(element, fireEvent),
+    ...(bindElementToEvents(element, fireEvent) as any), // could not type this :/
     ...bindElementToMethods(element, helpers),
     ...bindElementToMethods(element, customMethods)
-  }
+  } as BoundType[] // would be nicer without this cast
 }
```

Comment:
When this method is called from the outside it always return an Array. The other values (a single element or null) are returned only from deep inside the recursion, so I applied this hacky solution so it looks good from the outside and does not need a large refactoring